pkg/systemd/services/service-details.jsx

```diff
@@ -576,3 +655,26 @@ ServiceDetails.propTypes = {
     permitted: PropTypes.bool.isRequired,
     isValid: PropTypes.func.isRequired,
 };
+
+const DeleteModal = (props) => {
+    const [inProgress, setInProgress] = useState(false);
+    return (
+        <Modal isOpen
+               showClose={false}
+               position="top" variant="medium"
+               onClose={props.handleCancel}
+               title={cockpit.format(_("Confirm deletion of $0"), props.name)}
+               titleIconVariant="warning"
+               footer={<>
+                   <Button id="delete-timer-modal-btn" variant="danger" isDisabled={inProgress} isLoading={inProgress}
+                           onClick={() => { setInProgress(true); props.handleDelete().catch(() => setInProgress(false)) }}
```

Comment:
Hmm.. so on errors we keep the dialog open, but don't show the error? That's a bit awkward.

Reply:
Yeah, and hmm it doesn't really work...

Reply:
The promises in `deleteTimer` already catch their own errors so calling `unitAction`, `unitFileAction` will show an error in a new dialog quickly and then quickly navigate away as the promises succeed. (This is super annoying)